src/app/_services/api/user-actions.service.ts

```diff
@@ -0,0 +1,66 @@
+import {HttpClient, HttpParams} from '@angular/common/http';
+import {Injectable} from '@angular/core';
+import {environment} from '@environments/environment';
+import {Observable, of} from 'rxjs';
+import {catchError} from 'rxjs/operators';
+import {Action} from '@app/_models';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class UserActionsService {
+  private userActionUrl = new URL(
+    '/api/user-actions',
+    environment.apiBaseUrl
+  ).toString();
+
+  constructor(private http: HttpClient) {
+  }
+
+  getAllUserActions(options?): Observable<Action[]> {
+    const recent: boolean = options?.recent;
+    const params = new HttpParams().set('recent', `${recent ? recent : false}`);
+
+    return this.http
+      .get<Action[]>(this.userActionUrl, {params})
+      .pipe(
+        catchError(
+          this.handleError<Action[]>(
+            `getAllUserActions`
+          )
+        )
+      );
+  }
+
+  getUserActions(actionId: any, options?): Observable<Action> {
```

Comment:
Please rename this to getUserAction and the bulk one to getUserActions. Just for the sake of convention.